### PR TITLE
fix(starter-kit): Starter kit cleanup

### PR DESCRIPTION
**Pull request by hvanyo**
## Changes
Bring over components from test site:

- src/components/Editors
  - only change was removing block-quotes and strike-thru as the are not commonly used in site builds
- src/components/Flowcontainer
  - removed Accordions & Contentful variations
- src/components/Images
  - no change from test-site
- src/components/Layout
  - no change from test-site PR: https://github.com/johnsonandjohnson/Bodiless-JS/pull/455
- src/components/Menu 
   - from test-site PR: https://github.com/johnsonandjohnson/Bodiless-JS/pull/455 plus modified some colors and names of tokens to be generic 
- src/components/Page
  - no change from test-site
- src/components/Tout
  - no change from test-site
- src/components/Elements.token.ts
  - brought over from test-site minus strikethru 

Pages
- Homepage redone as flow-container
    - Added meta data
    - Added Data
- 404 redone as flow-container. 
    - Added meta data 
    - Added Data
 

Templates
 - Updated to use flow-container

Gatsby config & Package Files
- Added default support of gatsby-plugin-canonical-urls & gatsby-plugin-sitemap

Tailwind Config 
- Brought over nice commenting from test-site and removed custom background color for burger-menu

Env file
- Changed port to use 8001.

Extra items from documentation/discussion that affect both Test & Starter Kit:
- Removed dependency on siteMetadata and made title & logo editable.
- SiteURL remains but moved to ENV site and shared with Canonical URLS.  So its set in one place instead of two.
- Switched menu, footer, & logo over to use withNodeKey to encourage that pattern.
- small refactor of GTM data layer to its own file 
- Updated favicon to use generic name 'favicon.png' and added manifest plugin to gatsby.config to not use settings from gatsby-theme-bodiless.
 - fixed linting issues

Update documentation:
- /Development/Guides/SiteBuildBasics?id=site-level-data
- /Development/Guides/BuildingSites/BuildingOverview
- /Development/Guides/BuildingSites/ComponentsStarterKit
- README

## Test Instructions

Recommend in both cases for testing:  to create sites outside of monorepo and bodiless pack gatsby-theme-bodiless:

- Test the Example Test Site
   - Logo is now editable image and should appear in both desktop and mobile version.
   - Site title and Site copyright is editable (only Site Copyright was editable before)
   - Confirm Logo, Menu, Site Title, Site Copyright are all saving in src/data/site
- Test the Starter Test Site
   - Header (Test that image logo is editable image, the image link is not editable and fixed to '/'
   - Menu (Menu/burger Menu behaves like test site)
   - Footer (Site title and Site copyright is editable)
   - Pages Exist
      - Home page with flowcontainer and components on it.
      - 404 page with flowcontainer and default 404 copy as previously defined.
   - Canonical URLS are populating on static.
   - Sitemap.xml is populating on static.
   - Flowcontainer works as expected for all components.
   - GTM is on and working correctly on static.
   - Meta is working with home & 404 pages
   - Favicon should be set.
 - Confirm starter backend runs on 8001

### CODE REVIEW

File: examples/starter/src/components/Layout/header.tsx [hunk 1]
```diff
@@ -52,11 +55,18 @@ const HeaderClean: FC<Props> = ({ components }) => {
...
       </MenuContainer>
     </Wrapper>
   );
 };
 
-const Header = designable(headerComponents)(HeaderClean);
+const asSiteHeader = flow(
+  designable(headerComponents),
+  withDesign({
+    Menu: withNodeKey('MainMenu', 'site'),
```
Comment (wodenx) [inline examples/starter/src/components/Layout/header.tsx:67]:
```suggestion
    Menu: withNodeKey({ nodeKey: 'MainMenu', nodeCollection: 'site' }),
```

File: packages/gatsby-theme-bodiless/gatsby-config.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ const plugins = [
       background_color: '#663399',
       theme_color: '#663399',
       display: 'minimal-ui',
-      icon: 'src/images/gatsby-icon.png', // This path is relative to the root of the site.
```
Comment (hvanyo) [inline packages/gatsby-theme-bodiless/gatsby-config.js:40]: @beliayeu  I was documenting how to change favicon for a site as I cleaned up the starter kit: https://github.com/johnsonandjohnson/Bodiless-JS/pull/482/files#diff-5a14da22b508e54cae848fc0ac859c17 and felt this was better to be generic filename and not related to gatsyb-icon.png.  Any concerns in doing this?

Comment (beliayeu) [reply packages/gatsby-theme-bodiless/gatsby-config.js:40]: @hvanyo, the concern is that it is a breaking change. if the icon is overridden on a site level (EMEA team probably did this) then the site level icon will be reverted to the default one defined in the theme

File: examples/starter/src/components/Layout/logo.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  Img,
+  replaceWith,
+  withDesign,
+} from '@bodiless/fclasses';
+import { asEditableImage } from '../Elements.token';
+
+type LogoComponents = {
+  SiteReturn: ComponentType<any>,
+  SiteLogo: ComponentType<any>,
+};
```
[the remark this answers is not in the document]
Comment (hvanyo) [reply examples/starter/src/components/Layout/logo.tsx]: @VancheeZze updated

**Review by konst8 — APPROVED** [review]

**Pull request merged**